### DISCUSSION TITLE
Package Naming

### DISCUSSION THREAD

**Discussion by michael-hawker** [CommunityToolkit/Windows#34] [Feedback]
You can find out about adding a NuGet feed to Visual Studio in our [wiki here](https://aka.ms/wct/wiki/previewpackages).

The new vNext packages are available on this feed here:

> https://pkgs.dev.azure.com/dotnet/CommunityToolkit/_packaging/CommunityToolkit-MainLatest/nuget/v3/index.json

There are two sets of packages for each component:

- `CommunityToolkit.Uwp.*` for UWP + WinUI 2 or Uno.UI
- `CommunityToolkit.WinUI.*` for Windows App SDK + WinUI 3 or Uno.WinUI

Currently each package is versioned as `8.0.0-beta.#`, see more information on [versioning feedback discussion](https://github.com/CommunityToolkit/Windows/discussions/33).

We need to have two packages for technical reasons for Uno Platform support. Let us know your thoughts in this space and how it relates to our [unified namespaces](https://github.com/CommunityToolkit/Windows/discussions/32). Is this understandable or confusing? Should we use a different pattern, etc...?

**Comment by Pdawg-bytes** [comment]
I think the unified namespaces idea is a good one, and this ties in pretty well with that. I think the current naming scheme fits well, and it matches with 1st party package names too, causing less confusion.

**Comment by bogdan-patraucean** [comment]
Why not splitting them like this:

- `CommunityToolkit.WinUI2.*` for UWP
- `CommunityToolkit.WinUI3.*` for Windows App SDK
- `CommunityToolkit.UnoUI.*` for Uno Platform with WinUI2
- `CommunityToolkit.UnoWinUI.*` for UnoPlatform with WinUI3

OR just

- `CommunityToolkit.WinUI2.*`  for UWP + WinUI 2 or Uno.UI
- `CommunityToolkit.WinUI3.*` for Windows App SDK + WinUI 3 or Uno.WinUI

I would like the last one because it a bit more clear about what to expect, having Uwp and WinUI I think creates confusion.

**Reply by michael-hawker** [reply]
Technically, we only need two packages, so we wouldn't need to split into 4, we can support Uno.UI or Uno.WinUI out of the other places. Uno is the reason we need two, otherwise I believe the bug in WindowsAppSDK has been resolved that would allows us to ship a single package that supports UWP and WindowsAppSDK. But the WASM .NET Standard target is indistinguishable for Uno.UI vs. Uno.WinUI, so we need to have two.

Part of our reasoning of the current naming is that we figure we'll probably not support UWP packages _forever_. No idea, when we'd stop, probably wouldn't be for at least another 18 months at minimum if I had to guess now.

So, if we get to a point where we're only supporting WinUI 3 and our namespace (#32) is `CommunityToolkit.WinUI.*` then we'd reach a place where we have a single package and its naming matches the namespacing.

**Reply by bogdan-patraucean** [reply]
Makes sense considering the future plans.

**Comment by michael-hawker** [comment]
Last call for any additional feedback on this topic. From how things have been going though with our previews, I think this course of action makes sense to proceed with.
